source/MaterialXRuntime/RtValue.cpp

```diff
@@ -17,232 +19,74 @@
 namespace MaterialX
 {
 
-RtValue::RtValue(const Matrix33& v, RtValueStore<Matrix33>& store)
+RtValue::RtValue(const Matrix33& v, RtObject& owner)
 {
-    *reinterpret_cast<Matrix33**>(&_data) = store.alloc();
+    // Allocate storage for the value.
+    PrvAllocator& allocator = owner.data()->asA<PrvElement>()->getAllocator();
+    *reinterpret_cast<Matrix33**>(&_data) = allocator.allocType<Matrix33>();
+
+    // Copy the value.
     asMatrix33() = v;
 }
 
-RtValue::RtValue(const Matrix44& v, RtValueStore<Matrix44>& store)
+RtValue::RtValue(const Matrix44& v, RtObject& owner)
 {
-    *reinterpret_cast<Matrix44**>(&_data) = store.alloc();
+    // Allocate storage for the value.
+    PrvAllocator& allocator = owner.data()->asA<PrvElement>()->getAllocator();
+    *reinterpret_cast<Matrix44**>(&_data) = allocator.allocType<Matrix44>();
+
+    // Copy the value.
     asMatrix44() = v;
 }
 
-RtValue::RtValue(const string& v, RtValueStore<string>& store)
+RtValue::RtValue(const string& v, RtObject& owner)
 {
-    string* ptr = store.alloc();
-    *ptr = v;
-    asPtr() = ptr;
-}
+    // Allocate storage for the value.
+    PrvAllocator& allocator = owner.data()->asA<PrvElement>()->getAllocator();
+    *reinterpret_cast<string**>(&_data) = allocator.allocType<string>();
```

Comment:
This looks cleaner, but still feels strange to have a explicit memory allocation pass as an argument.

Reply:
Re: _"memory allocation pass as an argument"_ - Do you mean passing the RtObject owner as an argument?

It's a similar design as how we pass the parent element then creating new child elements. Normally you will not use these constructors directly, instead the idea is that you will use either RtValue::createNew() or even better RtTypeDef::createValue().

So either the first or second line here:
```
RtValue value = RtValue::createNew(RtType::FLOAT, parentObj);
RtValue value = floatType->createValue(parentObj)
```
Which is very similar to how element objects are created and placed into a parent.